dipy/reconst/dti.py

```diff
@@ -1801,90 +1850,102 @@ def restore_fit_tensor(design_matrix, data, sigma=None, jac=True,
         model_S0 = np.empty((flat_data.shape[0], 1))
     for vox in range(flat_data.shape[0]):
         if np.all(flat_data[vox] == 0):
-            raise ValueError("The data in this voxel contains only zeros")
+            raise ValueError("The data in this voxel contains only zeros") # FIXME: what to do if this is true? Crashing the code seems wrong
 
-        start_params = ols_params[vox]
-        # Do nlls using sigma weighting in this voxel:
-        if jac:
-            this_param, status = opt.leastsq(_nlls_err_func, start_params,
-                                             args=(design_matrix,
-                                                   flat_data[vox],
-                                                   'sigma',
-                                                   sigma),
-                                             Dfun=_nlls_jacobian_func)
-        else:
-            this_param, status = opt.leastsq(_nlls_err_func, start_params,
-                                             args=(design_matrix,
-                                                   flat_data[vox],
-                                                   'sigma',
-                                                   sigma))
+        try:  # FIXME: is this in correct place? if crashing on GMM, would we want to default to NL, rather than OLS??
```

Comment:
okay, I have fixed this to use the original NL result (if we managed to calculate it), or the OLS result if the original NL failed (same checks as in the nls fitting function)

Reply:
I have left this alone for now. Perhaps we should default for NLLS sigma-weighted solution if GMM fails, provided the initial fit for NLLS sigma-weighted solution did not fail, but since the GMM usually fails due to bad sigma choice, it is not clear to me that the NLLS is a good default choice after all. Suggest to leave alone for now. The option `fail_is_nan` and new user warnings should help.